### ISSUE TITLE
Convenience: add command to `fixup` current changes into patch on top

### ISSUE THREAD

**Issue by jazzdragon** [uptech/git-ps-rs#162]
Many times I just forget something small, or a test fails in my integration hook, and I just want to add some change to my latest commit. Would be nice to have a command to rebase the current changes straight into the latest patch, instead of having to commit and go through the interactive rebase.

**Comment by ctsstc** [comment]
It could also be nice to allow an optional index that would fixup the currently staged files into the index.

**Comment by Alizter** [comment]
What about `git commit --amend`?

**Comment by drewdeponte** [comment]
@Alizter we have `gps a` which is convenience for `git commit --amend`.

**Comment by Alizter** [comment]
@drewdeponte Sorry I didn't mean if there is something like that in gps, I thought I understood the issue as wanting what `git commit --amend` does.

**Comment by drewdeponte** [comment]
Yeah, I generally do the `gps a` when in this scenario. Sometimes I end of up in cases where I have another patch like this scenario is leading to. In that case I do as it says and I `gps rebase` to do an interactive rebase and squash/fixup the patch.

I think having a convenience method specifically to fixup a patch into the top most patch is pretty low on my priority list.

@ctsstc maybe you have some workflow differences where this scenario is happening regularly for you? I would love to get a better understanding of how you are regularly getting into these scenarios as I generally don't. I am generally using `gps a` or I am using `gps rebase` with an `edit` mark and then doing a `gps a` to add stuff to a patch in the middle of the stack.

**Comment by Alizter** [comment]
Also the editor you use for rebasing can make a lot of difference. I use emacs with magit which makes rebasing quite a pleasant experience. This is my only use for emacs.

**Comment by ctsstc** [comment]
I think my scenario where I need to fixup a patch that isn't on the top of the stack is when I haven't yet integrated a patch and I later find I need to refactor something like the interface in the older patch, so I end up needing to fixup against that patch. I guess I can do an interactive rebase and edit the patch, but most of the times I end up making a "fixup commit" and running the interactive rebase fixing up the patch that way.

**Comment by jazzdragon** [comment]
I also sometimes run into the scenario @ctsstc mentioned. Sometimes I see I need to make an update to a patch in the middle of my stack so I start with a rebase to edit that patch. Sometimes however, I either not think of it until after I made the code changes, or I have other changes that are unrelated but not committed yet.

Usually in that scenario I make a "fix" commit and then rebase it into place, but a convenience to throw that change into a patch index would be very useful to me.

I think the way to do so would have to be automating the process of:
- committing the staged changes
- rebase the commit into place with autostash
- maybe fail if we get conflicts? not sure.

Happy to tackle this if we're into the idea

**Comment by drewdeponte** [comment]
Git itself has the `git commit --fixup` command. The current development version of `gps` supports the `gps sha` command so that you can use it in composition with other commands, like `git commit --fixup`. For example, `git commit --fixup=amend:$(gps sha 4)` would do a `git commit --fixup=amend` for the commit pointed by patch index 4.

Anyways, the idea is that you can use Git aliases or shell aliases to codify the combinations that you use in your workflow. This is because maintaining everyones (very different workflows & edge cases, seems like a loosing battle).

See #266 for details and discussion around this and other examples.
